data-model/src/main/java/io/micronaut/data/model/schema/Table.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.model.schema;
+
+import io.micronaut.core.annotation.Internal;
+
+import java.util.List;
+
+/**
+ * The table definition. The information is extracted from the {@link io.micronaut.data.model.PersistentEntity}.
+ *
+ * @param schema The schema name, not required
+ * @param name The table name
+ * @param primaryKeyColumns The list of primary key columns, can be null or empty
+ * @param columns The list of columns
+ * @param sequences The list of table sequences, can be null or empty
+ *
+ * @author radovanradic
+ * @since 4.13.0
+ */
+@Internal
+public record Table(
```

Comment:
maybe rename to `TableDefinition`

Reply:
Changed all to have `Definition` suffix. Will need another `ColumnSomething` to map from the table columns metadata but will figure out different name.